rasa/core/channels/socketio.py

```diff
@@ -119,7 +119,7 @@ async def send_custom_json(
 
         await self.sio.emit(self.bot_message_evt, **json_message)
 
-    async def send_attachment(
+    async def send_attachment(  # type: ignore[override]
```

Comment:
looking at the different channels, I was wondering if actually all the channels support attachments, or if the method is just overridden because it needs to be. Do you know? If so, maybe using a mixin could help us here 🤔

Reply:
I've just checked all the channels, most don't implement attachments, apart from Hangouts, Rocketchat, SocketIO and Slack (the latter 2 being the one that violate the superclass type).

What's interesting though is that in the socketio [docs](https://python-socketio.readthedocs.io/en/latest/api.html?highlight=asyncserver#socketio.AsyncServer.emit), the data that gets sent can be of type `str`, but I'm a bit reluctant to change this to `str` in the super-class, maybe I can make the superclass accept Dict too, apart from `str`?
```
data – The data to send to the client or clients. Data can be of type str, bytes, list or dict. To send multiple arguments, use a tuple where each element is of one of the types indicated above.
```

Reply:
🤔 yeah I think it'd be a bit more work and would probably need more refactoring. Let's keep the `type: ignore` comment for now